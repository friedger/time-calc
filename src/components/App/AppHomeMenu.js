import React from 'react';
import autoBind from 'react-autobind';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import { withStyles } from '@material-ui/core';
import { navigateToApp } from '../../logic/actions/actions';
import { withRouter } from 'react-router-dom';

const styles = () => ({
  icon: {},
  primary: {},
});

class AppHomeMenu extends React.Component {
  state = {};

  static propTypes = {
    classes: PropTypes.object.isRequired,
    onHome: PropTypes.any,
    history: PropTypes.any,
  };

  constructor(props) {
    super(props);
    autoBind(this);
  }

  handleClick = () => {
    this.props.onHome(this.props.history);
  };

  render() {
    return (
      <IconButton color="inherit" onClick={this.handleClick} aria-label="Close">
        <CloseIcon />
      </IconButton>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onHome: history => dispatch(navigateToApp(history)),
  };
};

export default withRouter(connect(null, mapDispatchToProps)(withStyles(styles)(AppHomeMenu)));
