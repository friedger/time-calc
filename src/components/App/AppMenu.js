import React from "react";
import autoBind from "react-autobind";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import ShareIcon from "@material-ui/icons/Share";
import SettingsIcon from "@material-ui/icons/Settings";
import { withStyles, Dialog, DialogTitle } from "@material-ui/core";
import Button from "@material-ui/core/Button";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import {
  requestApproval,
  navigateToProjects
} from "../../logic/actions/actions";
import { withRouter } from "react-router-dom";

const ITEM_HEIGHT = 48;
const MENU_PROJECTS = "projects";
const MENU_REQUEST_APPROVAL = "request_approval";
const MENU_SHARE = "share";

const styles = () => ({
  icon: {},
  primary: {}
});

class AppMenu extends React.Component {
  state = {
    anchorEl: null,
    pickContact: false,
    customerCopyDialogOpen: false
  };

  static propTypes = {
    classes: PropTypes.object.isRequired,
    onRequestApproval: PropTypes.any,
    onProjectSettings: PropTypes.any,
    history: PropTypes.any,
    signedIn: PropTypes.bool.isRequired,
    hasCustomer: PropTypes.bool.isRequired,
    currentProject: PropTypes.object,
    customerCopySaved: PropTypes.bool,
    customerCopyUrl: PropTypes.string,
    shareUrl: PropTypes.string
  };

  constructor(props) {
    super(props);
    autoBind(this);
  }

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleMenuItemClick = (event, item) => {
    const { currentProject } = this.props;

    if (item === MENU_SHARE) {
      try {
        // eslint-disable-next-line no-console
        console.log("nav.share " + navigator.share);
        navigator.share({ title: "OI Timesheet", url: "" });
      } catch (err) {
        // eslint-disable-next-line no-console
        console.error("Share failed:", err.message);
        alert("Couldn't share (" + err + ")");
      }
    } else if (item === MENU_REQUEST_APPROVAL) {
      if (currentProject && currentProject.customer) {
        this.handleDialogClickOpen();
        this.props.onRequestApproval(currentProject.customer);
      }
    } else if (item === MENU_PROJECTS) {
      this.props.onProjectSettings(this.props.history);
    }
    this.handleClose();
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  handleDialogClickOpen = () => {
    this.setState({ customerCopyDialogOpen: true });
  };

  handleDialogClose = () => {
    this.setState({ customerCopyDialogOpen: false });
  };

  render() {
    const {
      classes,
      signedIn,
      hasCustomer,
      customerCopySaved,
      shareUrl
    } = this.props;
    const { anchorEl } = this.state;
    const open = Boolean(anchorEl);

    /* Hide menu in PublicHomePage
    if (match.path === "/") {
      return null;
    }
    */

    return (
      <div>
        <IconButton
          aria-label="More"
          aria-owns={open ? "app-menu" : undefined}
          aria-haspopup="true"
          onClick={this.handleClick}
        >
          <MoreVertIcon />
        </IconButton>
        <Menu
          id="app-menu"
          anchorEl={anchorEl}
          open={open}
          onClose={this.handleClose}
          PaperProps={{
            style: {
              maxHeight: ITEM_HEIGHT * 4.5,
              width: 200
            }
          }}
        >
          <MenuItem
            // eslint-disable-next-line no-console
            onClick={event => {
              this.handleMenuItemClick(event, MENU_PROJECTS);
            }}
          >
            <ListItemIcon className={classes.icon}>
              <SettingsIcon />
            </ListItemIcon>
            <ListItemText
              classes={{ primary: classes.primary }}
              inset
              primary="Project settings"
            />
          </MenuItem>
          {signedIn && hasCustomer && (
            <MenuItem
              // eslint-disable-next-line no-console
              onClick={event => {
                this.handleMenuItemClick(event, MENU_REQUEST_APPROVAL);
              }}
            >
              <ListItemIcon className={classes.icon}>
                <ShareIcon />
              </ListItemIcon>
              <ListItemText
                classes={{ primary: classes.primary }}
                inset
                primary="Share with customer"
              />
            </MenuItem>
          )}
          {navigator.share && (
            <MenuItem
              // eslint-disable-next-line no-console
              onClick={event => {
                this.handleMenuItemClick(event, MENU_SHARE);
              }}
            >
              <ListItemIcon className={classes.icon}>
                <ShareIcon />
              </ListItemIcon>
              <ListItemText
                classes={{ primary: classes.primary }}
                inset
                primary="Share app"
              />
            </MenuItem>
          )}
        </Menu>
        <Dialog
          open={this.state.customerCopyDialogOpen}
          onClose={this.handleDialogClose}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogTitle id="alert-dialog-title">
            {"Customer Copy"}
          </DialogTitle>
          {customerCopySaved && (
            <DialogContent>
              <DialogContentText id="alert-dialog-description">
                A copy has been stored for your customer/reviewer.                
              </DialogContentText>
              <DialogContentText id="alert-dialog-description">
                Please share with <b>{this.props.currentProject.customer}</b> the link below <br/>
                <a href={shareUrl}>{shareUrl}</a>               
              </DialogContentText>
            </DialogContent>
          )}
          {!customerCopySaved && (
            <DialogContent>
              <DialogContentText id="alert-dialog-description">
                Copy for your customer is currently generated.... please wait!
              </DialogContentText>
            </DialogContent>
          )}
          {customerCopySaved && (
            <DialogActions>
              <Button
                onClick={this.handleDialogClose}
                color="primary"
                autoFocus
              >
                Ok
              </Button>
            </DialogActions>
          )}
        </Dialog>
      </div>
    );
  }
}

const mapStateToProps = state => {
  let shareUrl;
  if (state.projectlist.currentProject && state.userProfile.user) {
    shareUrl = window.origin + "/#/shared/" + state.userProfile.user.username + "/" + state.projectlist.currentProject.id + "/" + state.projectlist.currentProject.filename;
  }
  return {
    currentProject: state.projectlist.currentProject,
    signedIn: !!state.userProfile.user && !state.userProfile.userMessage,
    hasCustomer: !!state.projectlist.currentProject.customer,
    customerCopySaved: !!state.syncState.customerCopySaved,
    customerCopyUrl: state.syncState.customerCopyUrl,
    shareUrl
  };
};
const mapDispatchToProps = dispatch => {
  return {
    onRequestApproval: userId => dispatch(requestApproval(userId)),
    onProjectSettings: history => dispatch(navigateToProjects(history))
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(withStyles(styles)(AppMenu))
);
