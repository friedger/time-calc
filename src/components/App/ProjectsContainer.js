import React from "react";
import Grid from "@material-ui/core/Grid";
import MomentUtils from "material-ui-pickers/utils/moment-utils";
import MuiPickersUtilsProvider from "material-ui-pickers/utils/MuiPickersUtilsProvider";

import moment from "moment";

import ProjectForm from "../ProjectForm/ProjectForm";


const ProjectsContainer = () => (<div>
              <Grid item xs={12}>
                <MuiPickersUtilsProvider utils={MomentUtils} moment={moment}>
                  <ProjectForm />
                </MuiPickersUtilsProvider>
              </Grid>              
            </div>
);
export default ProjectsContainer;