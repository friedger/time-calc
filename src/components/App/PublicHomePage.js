import React from "react";
import Typography from "@material-ui/core/Typography";

const launcherStyle = {
  width: "100vw",
  height: "50vh",
  alignContent: "center",
  alignItems: "center",
  boxSizing: "border-box",
  display: "flex",
  flexDirection: "row",
  flexWrap: "nowrap",
  justifyContent: "center",
  textAlign: "center"
};
const PublicHomePage = () => {
  return (
    <div style={launcherStyle}>
      <div>
        <div>
          <img height="100vh" width="100vw" src="/android-chrome-512x512.png" />
        </div>

        <Typography variant="headline">
          Be more productive! Track your time spent on projects and clients.
        </Typography>

        <Typography variant="body1">
          Your data never get out of your control.
        </Typography>

        <Typography variant="body1">
          Works offline. Wherever and whenever you need.
        </Typography>

        <Typography variant="body1">
          Across devices with <a href="https://blockstack.org"><b>Blockstack</b></a>.
        </Typography>
        <div>
          <a href="/#/app">
            <img alt="Launch as Web App" src="pwa.png" height="64vh" />
          </a>
        </div>
      </div>
    </div>
  );
};

export default PublicHomePage;
