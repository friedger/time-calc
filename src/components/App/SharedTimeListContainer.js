import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import TimeList from "../TimeList/TimeList";

import { loadSharedTimes } from "../../logic/actions/actions";

class SharedTimeListContainer extends Component {
  static propTypes = {
    match: PropTypes.any,
    loadTimes: PropTypes.func,
    timesLoaded: PropTypes.bool,
    message: PropTypes.string,
    project: PropTypes.object,
    owner: PropTypes.string,
    signedIn: PropTypes.bool
  };

  componentDidMount() {
    // eslint-disable-next-line no-console
    if (this.props.signedIn) {
      this.props.loadTimes();
    }
  }

  componentDidUpdate(prevProps) {
    if (!prevProps.signedIn && this.props.signedIn) {
      this.props.loadTimes();
    }
  }

  render() {
    const { match, timesLoaded, message, project, owner } = this.props;

    // eslint-disable-next-line no-console
    console.log("params", match.params);
    return (
      <div style={{ margin: 8 }}>
        <Grid container>
          <Grid item xs={12}>
            <Typography variant="h3">
              Shared Timesheet {owner && <>by {owner}</>}
            </Typography>
            <Typography variant="h4">{project.title || project.id}</Typography>
          </Grid>
          <Grid item xs={12}>
            <Paper>
              {timesLoaded && <TimeList readOnly="true" />}
              {!timesLoaded && (
                <Typography variant="body1">{message}</Typography>
              )}
            </Paper>
          </Grid>
        </Grid>
      </div>
    );
  }
}

export default connect(
  state => {
    let signedIn = !!state.userProfile.user && !state.userProfile.userMessage;
    let message;
    if (!signedIn) {
      message = "Please sign in to see shared timesheet.";
    } else {
      message = state.syncState.error || "Loading ...";
    }
    return {
      signedIn,
      timesLoaded:
        !!state.sharedTimesheet.times && state.sharedTimesheet.times.length > 0,
      project: state.sharedTimesheet.project || {},
      owner: state.sharedTimesheet.owner,
      message
    };
  },
  (dispatch, ownProps) => {
    const user = ownProps.match.params.user;
    const projectId = ownProps.match.params.project;
    const filename = ownProps.match.params.file;
    return {
      loadTimes: () => dispatch(loadSharedTimes(user, projectId, filename))
    };
  }
)(SharedTimeListContainer);
