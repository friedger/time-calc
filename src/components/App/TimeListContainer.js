import React from "react";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import MomentUtils from "material-ui-pickers/utils/moment-utils";
import MuiPickersUtilsProvider from "material-ui-pickers/utils/MuiPickersUtilsProvider";

import moment from "moment";

import Form from "../Form/Form";
import TimeList from "../TimeList/TimeList";


const TimeListContainer = () => (<div>
              <Grid item xs={12}>
                <MuiPickersUtilsProvider utils={MomentUtils} moment={moment}>
                  <Form />
                </MuiPickersUtilsProvider>
              </Grid>
              <Grid item xs={12}>
                <Paper>
                  <TimeList />
                </Paper>
              </Grid>
            </div>
);
export default TimeListContainer;