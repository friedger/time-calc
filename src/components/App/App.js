import React from "react";
import PropTypes from "prop-types";
import { HashRouter, Route } from "react-router-dom";

import Grid from "@material-ui/core/Grid";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import {
  MuiThemeProvider,
  createMuiTheme,
  withStyles,
  responsiveFontSizes
} from "@material-ui/core/styles";
import blue from "@material-ui/core/colors/blue";

import { Provider, connect } from "react-redux";
import { withRouter } from "react-router-dom";
import createStore from "./../../logic/store";

import Syncer from "../Syncer/Syncer";
import UserProfile from "../UserProfile/UserProfile";
import AppMenu from "./AppMenu";

import PublicHomePage from "./PublicHomePage";
import TimeListContainer from "./TimeListContainer";
import ProjectsContainer from "./ProjectsContainer";
import SharedTimeListContainer from "./SharedTimeListContainer";
import AppHomeMenu from "./AppHomeMenu";

const theme = responsiveFontSizes(
  createMuiTheme({
    palette: {
      primary: blue
    },
    typography: {
      fontDisplay: "swap"
    }
  })
);

const styles = () => ({
  root: {
    flexGrow: 1
  },
  title: {
    flexGrow: 1
  },
  footer: {
    backgroundColor: theme.palette.background.paper,
    marginTop: theme.spacing(8),
    padding: `${theme.spacing(6)}px 0`
  }
});

const BareOIAppBar = ({
  classes,
  currentProject,
  location,
  signedOut,
  sharedSheet
}) => {
  let title;
  let app = location.pathname !== "/";
  let projectDetails = location.pathname.startsWith("/project");
  if (app && !!currentProject && !!currentProject.title && !sharedSheet) {
    title = <div>OI Timesheet - {currentProject.title}</div>;
  } else {
    title = <div>OI Timesheet</div>;
  }
  return (
    <AppBar position="static" color="primary">
      <Toolbar>
        {projectDetails && <AppHomeMenu />}
        <Typography variant="h3" color="inherit" className={classes.title}>
          {title}
        </Typography>
        <UserProfile />
        {app && !signedOut && <Syncer />}
        {app && !projectDetails && <AppMenu />}
      </Toolbar>
    </AppBar>
  );
};

BareOIAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
  currentProject: PropTypes.object,
  match: PropTypes.any,
  location: PropTypes.any,
  signedOut: PropTypes.bool,
  history: PropTypes.object,
  sharedSheet: PropTypes.bool
};

const OIAppBar = withRouter(
  connect((state, ownProps) => {
    const sharedSheet =
      ownProps.location &&
      ownProps.location.pathname &&
      ownProps.location.pathname.startsWith("/shared");

    return {
      currentProject: state.projectlist.currentProject,
      signedOut: !state.userProfile.user && !state.userProfile.userMessage,
      sharedSheet
    };
  })(withStyles(styles)(BareOIAppBar))
);

const App = ({ classes }) => (
  <Provider store={createStore()}>
    <HashRouter>
      <MuiThemeProvider theme={theme}>
        <Grid container className={classes.root} spacing={2}>
          <OIAppBar />
          <Route path="/app" component={TimeListContainer} />
          <Route path="/project/:project" component={ProjectsContainer} />
          <Route
            path="/shared/:user/:project/:file"
            component={SharedTimeListContainer}
          />
          <Route path="/" exact component={PublicHomePage} />
        </Grid>
        {/* Footer */}
        <footer className={classes.footer} align="center">
          <Grid container spacing={2}>
            <Grid item xs={12} sm={6}>
              <Typography
                variant="body1"
                align="center"
                color="textSecondary"
                component="p"
              >
                About
              </Typography>
              <Typography variant="body2" align="center" gutterBottom>
                made by OpenIntents.org
              </Typography>
              <Typography variant="body2" align="center" color="textSecondary">
                <a href="https://github.com/friedger/time-calc">
                  Open Source And Free!
                </a>
                <br />
                Version {process.env.REACT_APP_VERSION}
                <br />
                <a href="https://github.com/friedger/time-calc/releases">
                  What is new
                </a>
                <br />
                <a href='https://github.com/friedger/time-calc/issues?q=is%3Aissue+is%3Aopen+label%3A"bug"'>
                  Known issues
                </a>
              </Typography>
            </Grid>

            <Grid item xs={12} sm={6}>
              <Typography
                variant="body1"
                align="center"
                color="textSecondary"
                component="p"
              >
                Contact
              </Typography>
              <Typography
                variant="body2"
                align="center"
                color="textSecondary"
                component="p"
              >
                <a href="https://app.trymumble.com/join/public/7df9971ea843-4c96-a86c-9acd67647c97">
                  Mumble
                </a>
                <br />
                <a href="mailto:support@openintents.org">Email</a>
              </Typography>
            </Grid>
          </Grid>
        </footer>
        {/* End footer */}
      </MuiThemeProvider>
    </HashRouter>
  </Provider>
);

App.propTypes = {
  classes: PropTypes.object.isRequired,
  currentProject: PropTypes.object
};

export default withStyles(styles)(App);
