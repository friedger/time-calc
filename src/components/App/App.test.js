import React from 'react'
import App from './App'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import {
  MuiThemeProvider
} from '@material-ui/core/styles'

import 'jest-enzyme'
import { TimeHelper } from '../../logic/helpers'

describe('Component - App', () => {
  it('renders correctly', () => {
    const component = shallow(<App />).dive()

    const regex = new RegExp(TimeHelper.now(), 'g')
    expect(component.html().replace(TimeHelper.today(), 'TODAY').replace(regex, 'NOW')).toMatchSnapshot()
  })

  it('wraps everything with redux', () => {
    const component = shallow(<App />).dive()

    expect(component.find(Provider)).toHaveProp('store')
  })

  it('wraps everything with MUI Theme', () => {
    const component = shallow(<App />).dive()

    expect(component.find(MuiThemeProvider)).toHaveProp('theme')
  })
})
