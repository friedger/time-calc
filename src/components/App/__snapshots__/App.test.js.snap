// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - App renders correctly 1`] = `"<div class=\\"MuiGrid-container-2 MuiGrid-spacing-xs-16-24 App-root-1\\"><header class=\\"MuiPaper-root-101 MuiPaper-elevation4-107 MuiAppBar-root-93 MuiAppBar-positionStatic-97 MuiAppBar-colorPrimary-99\\"><div class=\\"MuiToolbar-root-128 MuiToolbar-gutters-129\\"><h2 class=\\"MuiTypography-root-130 MuiTypography-title-136 MuiTypography-colorInherit-149\\">Timesheet</h2></div></header><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40\\"> <form><div class=\\"MuiPaper-root-101 MuiPaper-elevation2-105 MuiPaper-rounded-102 Form-control-155\\"><div class=\\"MuiGrid-container-2 MuiGrid-spacing-xs-16-24 MuiGrid-justify-xs-center-19 Form-root-154\\"><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-47 MuiGrid-grid-md-2-56\\"><div class=\\"MuiFormControl-root-157 MuiFormControl-fullWidth-160\\"><label class=\\"MuiFormLabel-root-166 MuiInputLabel-root-161 MuiInputLabel-formControl-162 MuiInputLabel-animated-165 MuiInputLabel-shrink-164\\" data-shrink=\\"true\\">Start Time</label><div class=\\"MuiInput-root-171 MuiInput-fullWidth-178 MuiInput-formControl-172 MuiInput-underline-175 WithUtils-DateTextField--input-156\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-179\\" placeholder=\\"NOW\\"/><div class=\\"MuiInputAdornment-root-184 MuiInputAdornment-positionEnd-186\\" style=\\"display:flex\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-193 MuiIconButton-root-187\\" type=\\"button\\"><span class=\\"MuiIconButton-label-192\\"> <span class=\\"material-icons MuiIcon-root-196\\" aria-hidden=\\"true\\"> timer </span> </span><span class=\\"MuiTouchRipple-root-202\\"></span></button></div></div></div></div><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-47 MuiGrid-grid-md-2-56\\"><div class=\\"MuiFormControl-root-157 MuiFormControl-fullWidth-160\\"><label class=\\"MuiFormLabel-root-166 MuiInputLabel-root-161 MuiInputLabel-formControl-162 MuiInputLabel-animated-165 MuiInputLabel-shrink-164\\" data-shrink=\\"true\\">End Time</label><div class=\\"MuiInput-root-171 MuiInput-fullWidth-178 MuiInput-formControl-172 MuiInput-underline-175 WithUtils-DateTextField--input-156\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-179\\" placeholder=\\"NOW\\"/><div class=\\"MuiInputAdornment-root-184 MuiInputAdornment-positionEnd-186\\" style=\\"display:flex\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-193 MuiIconButton-root-187\\" type=\\"button\\"><span class=\\"MuiIconButton-label-192\\"> <span class=\\"material-icons MuiIcon-root-196\\" aria-hidden=\\"true\\"> timer </span> </span><span class=\\"MuiTouchRipple-root-202\\"></span></button></div></div></div></div><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-47 MuiGrid-grid-md-2-56\\"><div class=\\"MuiFormControl-root-157 MuiFormControl-fullWidth-160\\"><label class=\\"MuiFormLabel-root-166 MuiInputLabel-root-161 MuiInputLabel-formControl-162 MuiInputLabel-animated-165 MuiInputLabel-shrink-164\\" data-shrink=\\"true\\">Break Time</label><div class=\\"MuiInput-root-171 MuiInput-fullWidth-178 MuiInput-formControl-172 MuiInput-underline-175 WithUtils-DateTextField--input-156\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-179\\" placeholder=\\"00:00\\"/><div class=\\"MuiInputAdornment-root-184 MuiInputAdornment-positionEnd-186\\" style=\\"display:none\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-193 MuiIconButton-root-187\\" type=\\"button\\"><span class=\\"MuiIconButton-label-192\\"> <span class=\\"material-icons MuiIcon-root-196\\" aria-hidden=\\"true\\"> timer </span> </span><span class=\\"MuiTouchRipple-root-202\\"></span></button></div></div></div></div><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-47 MuiGrid-grid-md-2-56\\"><div class=\\"MuiFormControl-root-157 MuiFormControl-fullWidth-160\\"><label class=\\"MuiFormLabel-root-166 MuiInputLabel-root-161 MuiInputLabel-formControl-162 MuiInputLabel-animated-165 MuiInputLabel-shrink-164\\" data-shrink=\\"true\\">Day</label><div class=\\"MuiInput-root-171 MuiInput-fullWidth-178 MuiInput-formControl-172 MuiInput-underline-175 WithUtils-DateTextField--input-156\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-179\\" placeholder=\\"TODAY\\"/><div class=\\"MuiInputAdornment-root-184 MuiInputAdornment-positionEnd-186\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-193 MuiIconButton-root-187\\" type=\\"button\\"><span class=\\"MuiIconButton-label-192\\"> <span class=\\"material-icons MuiIcon-root-196\\" aria-hidden=\\"true\\"> event </span> </span><span class=\\"MuiTouchRipple-root-202\\"></span></button></div></div></div></div><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-47 MuiGrid-grid-md-2-56\\"><div class=\\"MuiFormControl-root-157 MuiFormControl-fullWidth-160\\"><label class=\\"MuiFormLabel-root-166 MuiInputLabel-root-161 MuiInputLabel-formControl-162 MuiInputLabel-animated-165 MuiInputLabel-shrink-164\\" data-shrink=\\"true\\">Duration</label><div class=\\"MuiInput-root-171 MuiInput-disabled-174 MuiInput-fullWidth-178 MuiInput-formControl-172 MuiInput-underline-175 WithUtils-DateTextField--input-156\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-required=\\"false\\" class=\\"MuiInput-input-179 MuiInput-disabled-174\\" disabled=\\"\\" placeholder=\\"00:00\\"/><div class=\\"MuiInputAdornment-root-184 MuiInputAdornment-positionEnd-186\\" style=\\"display:none\\" disabled=\\"\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-193 MuiIconButton-root-187\\" type=\\"button\\"><span class=\\"MuiIconButton-label-192\\"> <span class=\\"material-icons MuiIcon-root-196\\" aria-hidden=\\"true\\"> timer </span> </span><span class=\\"MuiTouchRipple-root-202\\"></span></button></div></div></div></div><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-47 MuiGrid-grid-md-2-56\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-193 MuiIconButton-root-187 MuiIconButton-colorPrimary-189\\" type=\\"submit\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-192\\"><i class=\\"material-icons\\">add</i></span><span class=\\"MuiTouchRipple-root-202\\"></span></button></div></div><input type=\\"hidden\\" name=\\"index\\" value=\\"\\"/></div></form> </div><div class=\\"MuiGrid-item-3 MuiGrid-grid-xs-12-40\\"><div class=\\"MuiPaper-root-101 MuiPaper-elevation2-105 MuiPaper-rounded-102\\"></div></div></div>"`;
